test/devices/temperature_control_test.py

```diff
@@ -88,3 +91,99 @@ def test_temperature_control_activate():
     assert temperature_controller.k_p == 0.09
     assert temperature_controller.t_i == 0.000001
     assert temperature_controller.t_d == 9
+
+
+def test_temperature_control_active_below_set_point():
+    """
+    The function to test the response of the temperature controller
+    when it is active and below the set point
+    """
+    temperature_controller = create_temperature_controller()
+
+    temperature_controller.temperature_probe.sensor.set_temperature(25)
+
+    temperature_controller.activate()
+    assert temperature_controller.control_active is True
+
+    temperature_controller.update()
+    assert temperature_controller.heater.value is True
+
+
+def test_temperature_control_active_above_set_point():
+    """
+    The function to test the response of the temperature controller
+    when it is active and above the set point
+    """
+    temperature_controller = create_temperature_controller()
+
+    temperature_controller.temperature_probe.sensor.set_temperature(35)
+
+    temperature_controller.activate()
+    assert temperature_controller.control_active is True
+
+    temperature_controller.update()
+    assert temperature_controller.heater.value is False
+
+
+def test_temperature_control_active_at_set_point():
+    """
+    The function to test the response of the temperature controller
+    when it is active and at the set point
+    """
+    temperature_controller = create_temperature_controller()
+
+    temperature_controller.temperature_probe.sensor.set_temperature(30)
+
+    temperature_controller.activate()
+    assert temperature_controller.control_active is True
+
+    temperature_controller.update()
+    assert temperature_controller.heater.value is False
+
+
+def test_temperature_control_inactive_below_set_point():
+    """
+    The function to test the response of the temperature controller
+    when it is inactive and below the set point
+    """
+    temperature_controller = create_temperature_controller()
+
+    temperature_controller.temperature_probe.sensor.set_temperature(25)
+
+    temperature_controller.deactivate()
```

Comment:
Does it make sense to only call deactivate? Should we also have a test that activates, confirms its active, then deactivates and confirms it deactivate? That seems closer to real workflow. I think it is valuable to have a test that only calls deactivate, but I would add one that is closer to the full workflow.

Reply:
Those tests are added now.